edsdme/scripts/utils.js

```diff
@@ -150,6 +150,73 @@ export function redirectLoggedinPartner() {
   window.location.assign(target);
 }
 
+export async function getRenewBanner() {
+  const programType = getCurrentProgramType();
+  const accountExpiration = getPartnerDataCookieValue(programType, 'accountexpiration');
+  if (!accountExpiration) return;
+
+  const expirationDate = new Date(accountExpiration);
+  const now = new Date();
+
+  let metadataKey;
+  let daysNum;
+
+  const differenceInMilliseconds = expirationDate - now;
+  const differenceInDays = Math.abs(differenceInMilliseconds) / (1000 * 60 * 60 * 24);
+  const differenceInDaysRounded = Math.floor(differenceInDays);
+
+  if (differenceInMilliseconds > 0) {
+    if (differenceInDaysRounded === 0) {
+      metadataKey = 'banner-account-expires-today';
+    } else if (differenceInDays < 30) {
+      metadataKey = 'banner-account-expires';
+      daysNum = differenceInDaysRounded;
+    } else {
+      return;
+    }
+  } else if (differenceInMilliseconds < 0) {
+    if (differenceInDays < 90) {
+      metadataKey = 'banner-account-suspended';
+      daysNum = 90 - differenceInDaysRounded;
+    } else {
+      return;
+    }
+  }
+
+  const config = getConfig();
+  const [, locale] = config.locale.ietf.toLowerCase().split('-');
+  const defaultPath = locale === 'us' ? `/edsdme/partners-shared/fragments/${metadataKey}` : `/${locale}/edsdme/partners-shared/fragments/${metadataKey}`;
+  const path = getMetadataContent(metadataKey) ? getMetadataContent(metadataKey) : defaultPath;
```

Comment:
`getMetadataContent` returns `undefined` if there is no meta tag with a given name, so we can shorten this to `getMetadataContent(metadataKey) ?? defaultPath`